### ISSUE TITLE
fann: uniform non-zero Fisher makes project_delta zero the entire update, silently

### ISSUE THREAD

**Issue by ohdearquant** [ohdearquant/lattice#1575]
> _Issue authored by Claude (Anthropic agent) on behalf of @ohdearquant._

`DiagonalFisher::project_delta` damps by `1 - F_i/F_max`, so a Fisher whose entries are all equal and non-zero scales **every** coordinate to zero. The caller gets `Ok(())` and the parameters do not move. There is no log line and no error.

```rust
// crates/fann/src/training/ewc.rs
let f_max = self.values.iter().copied().fold(0.0_f32, f32::max);
if f_max < 1e-8 { return; }                       // all-zero Fisher: identity
for (d, &v) in delta.iter_mut().zip(self.values.iter()) {
    *d *= (1.0 - v / f_max).max(0.0);
}
```

### Why this is reachable rather than theoretical

The one production caller is `one_gradient_step` in `crates/tune/src/lora/router_update.rs`, which projects a step's own delta through a Fisher that the same step just updated:

1. `raw_delta = after - before`
2. `approx_grad = -raw_delta / lr`
3. `fisher.observe_gradient(&approx_grad)`, i.e. `F_i <- decay*F_i + (1-decay)*g_i^2`
4. `fisher.project_delta(&mut raw_delta)`

On the first step the EMA starts at zero, so `F_i` is proportional to `raw_delta_i^2`. `F_max` therefore sits at the largest-magnitude component of the update, and that component is scaled to exactly zero — the parameter the step most wanted to move is the one guaranteed not to move. On later steps the EMA blends history, so this weakens to "whenever the current gradient dominates the EMA at its own argmax".

The full stall is the symmetric case: equal-magnitude gradients make every coordinate the argmax at once, so the whole update is discarded.

### Test coverage gap

`ewc_high_fisher_blocks` pins the single-coordinate version as intended behaviour, which is fine. `ewc_degenerate_fisher_is_identity` covers an all-**zero** Fisher, which takes the `f_max < 1e-8` early return and is a different code path. Nothing covers uniform **non-zero** Fisher, which is the case that loses the update. The guarded degenerate case is the harmless one.

### Proposed work

- A failing test for uniform non-zero Fisher, and a second arm asserting the first-step largest-component zeroing, both mutation-checked (revert the fix, `touch` the source, watch them fail).
- Decide the intended semantics. If the aim is "damp the most important coordinates", a scale that cannot reach zero for the whole vector is needed — e.g. normalising by a quantile rather than the max, or adding a floor — and whatever is chosen should be stated in `crates/fann/docs/training.md` next to the existing description.
- Separately: `docs/adr/ADR-079-adaptive-micro-lora-brain.md`'s ROUTE row describes this as "Fisher null-space projection". It is elementwise damping, not a projection onto a null space. Worth correcting in the same pass so the doc and the code agree.

Found by source-read while auditing the router-update path; the arithmetic above has not been executed against a build, which is what the proposed tests are for.

**Comment by ohdearquant** [comment]
> _Comment authored by Claude (Anthropic agent) on behalf of @ohdearquant._

The issue closes with "the arithmetic above has not been executed against a build, which is what the proposed tests are for." It has now been executed in an FP32 replica of the update rule, and the symmetric stall reproduces. Three things to add: a fully specified construction, the executed numbers, and an algebraic argument that settles the ADR-079 wording question independently of anyone's reading of the code.

### The construction that stalls

Linear gate, 384 inputs, two outputs. Zero weights and zero biases, zero context. Auxiliary and z losses disabled, learning rate `0.001`, Fisher initialised to zero with decay `0.99`. Reject one of the two tied arms. These are all accepted configurations.

Only the output biases have non-zero gradients, and by symmetry they are equal and opposite:

```
g      = (+0.5, -0.5)
F_new  = 0.01 * (0.25, 0.25) = (0.0025, 0.0025)
```

Both entries equal `F_max`, so both damping factors are `1 - F_i/F_max = 0`. Neither bias moves. Repetition preserves the symmetry, so every subsequent update is also zero.

### Executed

An FP32 replica of `observe_gradient` followed by `project_delta` was run for 200 feedback events with the above configuration. After 200 events the biases are still `(0, 0)`, while both Fisher entries have grown to approximately `0.216505`. So the importance estimate accumulates normally the whole time and the parameters never move — the state that would make this visible from outside is precisely the state that is not inspected.

The undamped positive control, which is what makes this a defect rather than a property of the configuration: with the projection removed, the first SGD step moves the biases by `(-0.0005, +0.0005)`. Same construction, same data, non-zero movement. A Rust version of both arms against the real API has been written but not compiled (the environment producing it had no Rust toolchain), so the executed evidence above is the replica, not the crate.

### The projection wording is decidable without reading intent

The last bullet of the issue proposes correcting ADR-079's "Fisher null-space projection". That correction is forced, not stylistic: a projection must be idempotent, and this map is not. For `F = (1, 2)`:

```
D   = diag(1 - 1/2, 1 - 2/2) = diag(0.5, 0)
D^2 = diag(0.25, 0)
D^2 != D
```

So it is not an orthogonal projection, and it constructs no basis that gradients or representations are preserved against — which is the property a null-space method is chosen for. A real subspace method retains an explicit basis. Worth stating that way in the doc correction, because "elementwise damping, not a projection" is a claim a reader can check in two lines rather than a matter of terminology.

### One suggestion on the proposed tests

The symmetric-stall arm is worth asserting over many steps rather than one. A single-step assertion of "delta is zero" passes for the harmless all-zero-Fisher case too, and the interesting property here is that the stall is **absorbing**: the Fisher keeps growing (to ~0.216505 by event 200) while the parameters stay put, so a test that only checks step 1 cannot distinguish "damped once" from "can never adapt again". Asserting non-movement at event 200 *together with* a grown Fisher pins the absorbing behaviour, and the undamped control is what keeps that arm mutation-sensitive.
